src/Teambuilder/damagecalc.cpp

```diff
@@ -0,0 +1,836 @@
+                                                ? (wonderroom ? ostats[4] : ostats[2]) : (wonderroom ? ostats[2] : ostats[4])) : (wonderroom ? ostats[4] : ostats[2]));
+    bool usingspecial = (category == Move::Special || move == Move::Psyshock || move == Move::Psystrike || move == Move::SecretSword);
+
+    // -1B. SANDSTORM //
+    if (gen >= 4 && usingspecial && weather == Weather::SandStorm && (otype1 == Type::Rock || otype2 == Type::Rock)) {
+        defense = applymod(defense, 0x1800);
+    }
+    // -1C. MARVEL SCALE //
+    if (oability == Ability::MarvelScale && ostatus != Pokemon::Fine) {
+        dmod = chainmod(dmod, 0x1800);
+    }
+    // -1D. FLOWER GIFT //
+    if (ui->oflowergift->isChecked() && weather == Weather::Sunny && usingspecial) {
+        dmod = chainmod(dmod, 0x1800);
+    }
+    // -1E. DEEPSEASCALE //
+    if (opoke == Pokemon::Clamperl && oitem == Item::DeepSeaScale && usingspecial) {
+        dmod = chainmod(dmod, 0x1800);
+    }
+    // -1F. METAL POWDER //
+    if (opoke == Pokemon::Ditto && oitem == Item::MetalPowder && !usingspecial) {
+        dmod = chainmod(dmod, 0x2000);
+    }
+    // -1G. EVIOLITE //
+    if (PokemonInfo::HasEvolutions(opoke.pokenum) && oitem == Item::Eviolite) {
+        dmod = chainmod(dmod, 0x1800);
+    }
+    // -1H. SOUL DEW //
+    if ((opoke == Pokemon::Latios || opoke == Pokemon::Latias) && oitem == Item::SoulDew && usingspecial) {
+        dmod = chainmod(dmod, 0x1800);
+    }
+
+    defense = applymod(defense, dmod);
+
+    // 0. BASE //
+    int bmod = 0x1000;
+
+    // 0A. TECHNICIAN //
+    if (myability == Ability::Technician && bp < 60) {
+        bmod = chainmod(bmod, 0x1800);
+    }
+    // 0B. FLARE BOOST //
+    if (myability == Ability::FlareBoost && category == Move::Special) {
+        bmod = chainmod(bmod, 0x1800);
+    }
+    // 0C. ANALYTIC // 0x14CD
+    // 0D. RECKLESS //
+    if (myability == Ability::Reckless && (MoveInfo::Recoil(move, m_currentGen) > 0 || move == Move::JumpKick || move == Move::HiJumpKick)) {
+        bmod = chainmod(bmod, 0x1333);
+    }
+    // 0E. IRON FIST //
+    if (myability == Ability::IronFist && MoveInfo::Flags(move, m_currentGen) & Move::PunchFlag) {
+        bmod = chainmod(bmod, 0x1333);
+    }
+    // 0F. TOXIC BOOST //
+    if (myability == Ability::ToxicBoost && category == Move::Physical) {
+        bmod = chainmod(bmod, 0x1800);
+    }
+    // 0G. RIVALRY //
+    if (myability == Ability::Rivalry) {
+        int rmod = 0x1000;
+
+        if (mygender == Pokemon::Neutral || ogender == Pokemon::Neutral) {
+            rmod = 0x1000;
+        } else if (mygender == ogender) {
+            rmod = 0x1400;
+        } else {
+            rmod = 0xC00;
+        }
+
+        bmod = chainmod(bmod, rmod);
+    }
+    // 0H. SAND FORCE //
+    if (myability == Ability::SandForce && (type == Type::Rock || type == Type::Ground || type == Type::Steel && weather == Weather::SandStorm)) {
```

Comment:
Change this line to
`if (myability == Ability::SandForce && (type == Type::Rock || type ==         Type::Ground || (type == Type::Steel && weather == Weather::SandStorm)))     {`
to silence a warning in Qt/Travis